gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/task/DagTask.java

```diff
@@ -42,11 +43,14 @@ public abstract class DagTask {
   @Getter public final DagActionStore.DagAction dagAction;
   private final MultiActiveLeaseArbiter.LeaseObtainedStatus leaseObtainedStatus;
   @Getter protected final DagManager.DagId dagId;
+  @Getter protected final DagNodeId dagNodeId;
 
   public DagTask(DagActionStore.DagAction dagAction, MultiActiveLeaseArbiter.LeaseObtainedStatus leaseObtainedStatus) {
     this.dagAction = dagAction;
     this.leaseObtainedStatus = leaseObtainedStatus;
     this.dagId = DagManagerUtils.generateDagId(dagAction.getFlowGroup(), dagAction.getFlowName(), dagAction.getFlowExecutionId());
+    this.dagNodeId = new DagNodeId(dagAction.getFlowGroup(), dagAction.getFlowName(),
+        Long.parseLong(dagAction.getFlowExecutionId()), dagAction.getFlowGroup(), dagAction.getJobName());
```

Comment:
DagNodeId cannot be returned from DagAction class due to circular dependency issue.
I need to move DagAction from gobblin-runtime to gobblin-service to be able to do that, which makes sense because DA belongs to gaas. Should I go ahead?

Reply:
cc @umustafi